### PR TITLE
Bring circulation up to date with recent core changes

### PR DESCRIPTION
**Pull request by leonardr**
This branch changes circulation to respect the merging of Collection.protocol and ExternalIntegration.provider into Collection.provider.

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def test_from_config(self):
             eq_(Configuration.DEFAULT_FACET, facet_config._default_facets)
 
         with temp_config() as config:
+            config[Configuration.POLICIES] = {}
```
Comment (leonardr) [inline tests/test_config.py]: Unrelated change that has been annoying me for a while. I run tests with an extremely minimal configuration that doesn't define POLICIES at all, which is legal but which broke this test.

File: api/admin/controller.py [hunk 2]
```diff
@@ -1071,26 +1072,26 @@ def admin_auth_services(self):
 
             return dict(
                 admin_auth_services=auth_services,
-                providers=ExternalIntegration.ADMIN_AUTH_PROVIDERS,
+                providers=ExternalIntegration.ADMIN_AUTH_PROTOCOLS,
             )
 
-        provider = flask.request.form.get("provider")
-        if not provider:
+        protocol = flask.request.form.get("provider")
```
Comment (leonardr) [inline api/admin/controller.py]: I didn't change `provider` to `protocol` in the web app because I think it makes sense to continue to call this a "provider" in user-facing applications. I can change it though if we want to be consistent.

Comment (aslagle) [inline api/admin/controller.py]: That's fine. This doesn't affect what's displayed to users, but if we did want to change it we'd also have to update the form in circulation-web.
